### ISSUE TITLE
LDAP backup server not called when LDAP main server fails

### ISSUE THREAD

**Issue by GitHubUser4234** [nextcloud/server#10160]
Hi,

#10031 does its job well, but unfortunately it seems to create another issue, which is that the LDAP backup server is not called when the LDAP main server fails. To reproduce, make sure that main and backup LDAP server settings point to LDAP servers that are accessible. Then change the value of the main server setting to some invalid domain/port. Now LDAP users cannot login to Nextcloud anymore. Then replace the apps/user_ldap/lib/Connection.php with a version before the #10031 patch, Nextcloud is working normally again.

Looks like in  apps/user_ldap/lib/Connection.php there is an infinite loop (`establishConnection()` => `bind()` => `getConnectionResource()` => `establishConnection()`) once the primary server becomes inaccessible.

Please have a look, thanks!

**Comment by juliushaertl** [comment]
I could reproduce this and I prepared a fix at #10227 

The infinite loop was caused by calling bind even if there were no connection details set. Without those there was no connection resource available so it tried to get those for the main server again by calling establishConnection().

@GitHubUser4234 Could you try if the following patch fixes the issue for you:

```
diff --git a/apps/user_ldap/lib/Connection.php b/apps/user_ldap/lib/Connection.php
index 039e46aeef..8bbe034585 100644
--- a/apps/user_ldap/lib/Connection.php
+++ b/apps/user_ldap/lib/Connection.php
@@ -564,8 +564,8 @@ class Connection extends LDAPUtility {
 				if (!$isOverrideMainServer) {
 					$this->doConnect($this->configuration->ldapHost,
 						$this->configuration->ldapPort);
+					return $this->bind();
 				}
-				return $this->bind();
 			} catch (ServerNotAvailableException $e) {
 				if(!$isBackupHost) {
 					throw $e;
```

**Comment by MorrisJobke** [comment]
Fixed by #10227
